### ISSUE TITLE
[Proposal]: Support `Picamera2`

### ISSUE THREAD

**Issue by roaldarbol** [abhiTronix/vidgear#342]
### Issue guidelines

- [X] I've read the [Issue Guidelines](https://abhitronix.github.io/vidgear/latest/contribution/issue/#submitting-an-issue-guidelines) and wholeheartedly agree.

### Issue Checklist

- [X] I have searched open or closed [issues](https://github.com/abhiTronix/vidgear/issues) and found nothing related to my idea.
- [X] I have read the [Documentation](https://abhitronix.github.io/vidgear/latest) and it doesn't mention anything about my idea.
- [X] To my best knowledge, my idea wouldn't break something for other users.

### Describe your Idea

With Bullseye, Raspberry Pi moved away from the legacy camera support to their own `libcamera`-fuelled `picamera2`. It aims at being a slot-in replacement for `picamera`. It is still possible to use the legacy `picamera`, but it's being deprecated and is slowly starting to break things. 

Although `picamera2` is still in beta, I'd propose beginning to implement it.

### Use Cases

Anyone using Raspberry Pi cameras, especially newer custom cameras (e.g. from Arducam).

### Any other Relevant Information?

The `picamera2`repo: https://github.com/raspberrypi/picamera2
Documentation: https://datasheets.raspberrypi.com/camera/picamera2-manual.pdf

I can begin slowly and create a pull request, hoping that it really is mostly a slot-in replacement. Would you prefer having an additional `PiGear2` class inside `pigear.py`, or a new `pigear2.py`?

**Comment by dante-biase** [comment]
Hi, we're eagerly waiting on PiCamera2 support. Any progress updates on this/is there any development code we can look at or contribute to?

**Comment by abhiTronix** [comment]
@dante-biase I just got time to work on this project again. I'll start shortly after pending commits are pushed.

**Comment by dante-biase** [comment]
Fantastic!

**Comment by iandol** [comment]
I hope this is still in scope, the new pi camera modules are really nice (autofocus, wideangle etc), so it would be great to see picamera2 integrated here...

**Comment by roaldarbol** [comment]
It would also be a major selling point, as they are still not supported in OpenCV natively yet (https://github.com/opencv/opencv/issues/21653).

**Comment by sourabmaity** [comment]
Waiting for Picamera2 update

**Comment by abhiTronix** [comment]
@sourabmaity Yeah I'm swamped with a ton of work right now and my schedule's pretty tight. However, I'm committed to work on this, so as soon as I get a breather, I'll definitely circle back and address this.

**Comment by abhiTronix** [comment]
Okay I've finally started working on this. Expect a update in few days.

**Comment by dante-biase** [comment]
Any update on this?

**Comment by abhiTronix** [comment]
@dante-biase It's here finally. :)

**Comment by dante-biase** [comment]
Alright!

**Comment by roaldarbol** [comment]
Awesome to hear!🚀

**Comment by abhiTronix** [comment]
**Hi, everyone. The vidgear `v0.3.3` with Picamera2 support is officially released: https://pypi.org/project/vidgear/**
